libtm build fail

Hello!

In libtm CMakeFile (libtm/resources/CMakeLists.txt) generates broken fw.h file.

The problem in lines 113-127
```
# Create empty output file
file(WRITE ${FW_OUTPUT_FILE_TMP} "")
file(APPEND ${FW_OUTPUT_FILE_TMP} "/*******************************************************************************\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "INTEL CORPORATION PROPRIETARY INFORMATION\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "Copyright(c) 2017 Intel Corporation. All Rights Reserved.\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "*******************************************************************************/\n\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "#ifndef target_h\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "#define target_h\n\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "#define FW_VERSION \"${FW_VERSION}\"\n\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "static uint32_t target_hex []  = {\n")
    
bin2h(${LIBTM_RESOURCES_DIR}/target.mvcmd 0 fw_ready_buffer)

file(APPEND ${FW_OUTPUT_FILE_TMP} "${fw_ready_buffer}\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "};\n")
file(APPEND ${FW_OUTPUT_FILE_TMP} "#endif\n")
```

Placing `bin2h(${LIBTM_RESOURCES_DIR}/target.mvcmd 0 fw_ready_buffer)` before `file(WRITE ${FW_OUTPUT_FILE_TMP} "")` fixed problem for me. (CMake version 3.13.1, Windows)

Thanks!

@ev-mp the generated fw.h file without strings:
```
*******************************************************************************
INTEL CORPORATION PROPRIETARY INFORMATION
Copyright(c) 2017 Intel Corporation. All Rights Reserved.
*******************************************************************************
#ifndef target_h
#define target_h
#define FW_VERSION <version_placeholder>
static uint32_t target_hex []  = {
```
So there are just syntax errors:

```
c:\local\vcpkg_temp\buildtrees\realsense2\src\v2.19.1-4ce227aaf9\third-party\libtm\libtm\src\fw.h(1): error C2059: syntax error: 'constant' (compiling source file C:\local\vcpkg_temp\buildtrees\realsense2\src\v2.19.1-4ce227aaf9\third-party\libtm\libtm\src\Manager.cpp) [C:\local\vcpkg_temp\buildtrees\realsense2\x64-windows-dbg\third-party\libtm\libtm\src\tm.vcxproj]
c:\local\vcpkg_temp\buildtrees\realsense2\src\v2.19.1-4ce227aaf9\third-party\libtm\libtm\src\fw.h(130405): error C2143: syntax error: missing ';' before '}' (compiling source file C:\local\vcpkg_temp\buildtrees\realsense2\src\v2.19.1-4ce227aaf9\third-party\libtm\libtm\src\Manager.cpp) [C:\local\vcpkg_temp\buildtrees\realsense2\x64-windows-dbg\third-party\libtm\libtm\src\tm.vcxproj]
c:\local\vcpkg_temp\buildtrees\realsense2\src\v2.19.1-4ce227aaf9\third-party\libtm\libtm\src\fw.h(130405): error C2059: syntax error: '}' (compiling source file C:\local\vcpkg_temp\buildtrees\realsense2\src\v2.19.1-4ce227aaf9\third-party\libtm\libtm\src\Manager.cpp) [C:\local\vcpkg_temp\buildtrees\realsense2\x64-windows-dbg\third-party\libtm\libtm\src\tm.vcxproj]
c:\local\vcpkg_temp\buildtrees\realsense2\src\v2.19.1-4ce227aaf9\third-party\libtm\libtm\src\fw.h(130406): fatal error C1020: unexpected #endif (compiling source file C:\local\vcpkg_temp\buildtrees\realsense2\src\v2.19.1-4ce227aaf9\third-party\libtm\libtm\src\Manager.cpp) [C:\local\vcpkg_temp\buildtrees\realsense2\x64-windows-dbg\third-party\libtm\libtm\src\tm.vcxproj]
```

target.mvcmd is not zero-sized

@dorodnic oh, I didn't know. I didn't try to build just main SDK, but I will try.

> @claudiofantacci you can try my [portfile](https://github.com/IntelRealSense/librealsense/files/3012451/portfile.cmake.txt) with disabled libtm, seems like it works for ver. 2.19.1 (tagged as release) and current dev branch. I mean it works even without suggested workaround.

Thanks @elevir, but it still does not work reliably. Here is the log using `ninja` under Windows:
```cmake
[70/107] Performing download step (git clone) for 'libusb'
FAILED: libusb-prefix/src/libusb-stamp/libusb-download
cmd.exe /C "cd /D C:\Users\cfantacci\GitHub\librealsense\buildprovaninja\third-party && "C:\Program Files\CMake\bin\cmake.exe" -P C:/Users/cfantacci/GitHub/librealsense/buildprovaninja/libusb-prefix/tmp/libusb-gitclone.cmake && "C:\Program Files\CMake\bin\cmake.exe" -E touch C:/Users/cfantacci/GitHub/librealsense/buildprovaninja/libusb-prefix/src/libusb-stamp/libusb-download"
fatal: could not create work tree dir 'libusb': Permission denied
fatal: could not create work tree dir 'libusb': Permission denied
fatal: could not create work tree dir 'libusb': Permission denied
-- Had to git clone more than once:
          3 times.
CMake Error at C:/Users/cfantacci/GitHub/librealsense/buildprovaninja/libusb-prefix/tmp/libusb-gitclone.cmake:66 (message):
  Failed to clone repository: 'https://github.com/libusb/libusb.git'
```

Please have a look at #3597 where I collected all the issues I found so far and possible solutions. Is a long post, but I think worth reading (or you can jump at the end with the proposed soutions 🤣).

--- a/.gitignore
+++ b/.gitignore
@@ -4,8 +4,12 @@ lib/
 ubuntu-xenial/
 ubuntu-xenial-hwe/
 
+# Docs
+doc/doxygen/html/
+*.html.lnk
+
 # CMake
-build/
+build*/
 connectivity_check
 
 # XCode
@@ -83,4 +87,3 @@ librealsense-log.txt
 *.json
 *.ini
 *.cxx
-

--- a/CMake/windows_config.cmake
+++ b/CMake/windows_config.cmake
@@ -8,9 +8,7 @@ macro(os_set_flags)
     # Makes VS15 find the DLL when trying to run examples/tests
     set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
     set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
-    # build with multiple cores
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP")
+
     ## Check for Windows Version ##
     if( (${CMAKE_SYSTEM_VERSION} EQUAL 6.1) OR (FORCE_WINUSB_UVC) ) # Windows 7
         message(STATUS "Build for Win7")
@@ -26,7 +24,17 @@ macro(os_set_flags)
         set(BACKEND RS2_USE_WINUSB_UVC_BACKEND)
     endif()
 
-    if (MSVC)
+    if(MSVC)
+        # Set CMAKE_DEBUG_POSTFIX to "d" to add a trailing "d" to library
+        # built in debug mode. In this Windows user can compile, build and install the
+        # library in both Release and Debug configuration avoiding naming clashes in the
+        # installation directories.
+        set(CMAKE_DEBUG_POSTFIX "d")
+
+        # build with multiple cores
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP")
+
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj /wd4819")
         set(LRS_TRY_USE_AVX true)
         add_definitions(-D_UNICODE)

